cmd/gen.go

```diff
@@ -88,6 +95,7 @@ func init() {
 	genFlags.StringVar(&projectId, "project-id", "", "Generate types from a project ID.")
 	genFlags.StringVar(&dbUrl, "db-url", "", "Generate types from a database url.")
 	genFlags.StringArrayVar(&schemas, "schema", []string{}, "Schemas to generate types for.")
+	genFlags.BoolVar(&detectO2oRelationships, "detect-o2o-relationships", true, "Detect one-to-one relationships compatible with PostgREST v10+. Only use in together with --db-url.")
```

Comment:
Shall we rename this to `--for-PostgREST-9`? I think it's more intuitive flag name because users may not be fully aware of the differences in features supported but they should know the major version of their self-hosted PostgREST.

Doing so also leaves us some room to add other backwards compatibility switches in the future under that flag. 

Note that the book condition is inverted if we rename like above, ie. default value is false.